src/main/java/org/saltyrtc/client/crypto/CryptoProvider.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2018 Threema GmbH
+ *
+ * Licensed under the Apache License, Version 2.0, <see LICENSE-APACHE file>
+ * or the MIT license <see LICENSE-MIT file>, at your option. This file may not be
+ * copied, modified, or distributed except according to those terms.
+ */
+package org.saltyrtc.client.crypto;
+
+import org.saltyrtc.client.annotations.NonNull;
+
+/**
+ * An abstraction over NaCl.
+ */
+public interface CryptoProvider {
+
+    // These will be the same for all implementations, since they're specified by NaCl.
+    int PUBLICKEYBYTES = 32;
+    int PRIVATEKEYBYTES = 32;
+    int SYMMKEYBYTES = 32;
+    int NONCEBYTES = 24;
+    int BOXOVERHEAD = 16;
+
+    /**
+     * Create a public/private key pair.
+     *
+     * @param publickey A byte array with size PUBLICKEYBYTES
+     * @param privatekey A byte array with size PRIVATEKEYBYTES
+     */
+    void genkeypair(@NonNull byte[] publickey, @NonNull byte[] privatekey);
```

Comment:
`genkeypair` -> `generateKeyPair`?

Reply:
Will change that. I used `genkeypair` since it matches the C function name, but there's not really a need for that :slightly_smiling_face: